### ISSUE TITLE
OST downloads from website: No file permissions 

### ISSUE THREAD

**Issue by simonnzg** [osTicket/osTicket#4703]
I downloaded OST (both 1.11 and 1.10) from the website this morning to do some testing and note that the manufactured zip file contents have no permissions.  I could sort-of make the download work by adding some defaults, but I still got errors.

![noperm](https://user-images.githubusercontent.com/524239/52471557-15564500-2b89-11e9-914e-2adae3b4146e.PNG)

Also, if I select some languages, the manufactured zip file contains the languages and the clean installation archive, which has the same name as the download bundle.  The first question when unzipping from a command line is whether I should overwrite the file I'm unzipping with the contents.

**Comment by ntozier** [comment]
I can confirm the permissions part.  Maybe it was zipped on Windows.  I would just chmod it.

**Comment by JediKev** [comment]
@simonnzg 

It was packaged on Linux from a fresh clone so not sure why the file permissions are borked. Will update the site soon.

Cheers.

**Comment by simonnzg** [comment]
Yes, no problem for me, but I guess a bit of a headscratcher for some.  :)
So, because the directories are already execute, it's into several iterations of the "ugo" chmod method to pull all the permissions and then put them back in the right order. You can't just run 755 or 744 on the lot as the files should be read only for Group and Other, but the folders need execute.  Such fun.
Building the permissions in properly in the first place would be better. Without the permission set all you get is a white screen when running the installer.

**Comment by simonnzg** [comment]
Hi @JediKev  - I just saw your comment.  Thanks!

**Comment by tricoos** [comment]
Still not fixed so I had to do this which worked for me:
`cd upload
find ./ -type f -perm 000 -print -exec chmod 644 {} \;
`
